pkg/ast/spl/spl.peg

```diff
@@ -951,6 +956,301 @@ TcOptionCMD <- option:("usenull" / "useother" / "nullstr" / "otherstr") {
     return optionStr, nil
 }
 
+SpanLenOption <- number:(FloatAsString / IntegerAsString) timeScale:(TimeScale)? {
+    if timeScale != nil {
+        num, err := strconv.ParseInt(number.(string), 10, 64)
+        if err != nil {
+            return nil, fmt.Errorf("SpanLength: Invalid num (%v): %v", number.(string), err)
+        }
+
+        switch timeScale.(utils.TimeUnit) {
+            case utils.TMDecisecond:
+                if 10%num != 0 || num >= 10 {
+                    return nil, fmt.Errorf("SpanLength: Invalid span for decisecond %v, span value must be less than 1 second and 1 second must be evenly divisble by span", num)
+                }
+            case utils.TMCentisecond:
+                if 100%num != 0 || num >= 100 {
+                    return nil, fmt.Errorf("SpanLength: Invalid span for centisecond %v, span value must be less than 1 second and 1 second must be evenly divisble by span", num)
+                }
+            case utils.TMMillisecond:
+                if 1000%num != 0 || num >= 1000 {
+                    return nil, fmt.Errorf("SpanLength: Invalid span for millisecond %v, span value must be less than 1 second and 1 second must be evenly divisble by span", num)
+                }
+        }
+
+        spanLength := &structs.BinSpanLength {
+            Num: float64(num),
+            TimeScale: timeScale.(utils.TimeUnit),
+        }
+
+        return spanLength, nil
+    } 
+
+    num, err := strconv.ParseFloat(number.(string), 64)
+    if err != nil {
+        return nil, fmt.Errorf("SpanLength: Invalid num (%v): %v", number.(string), err)
+    }
+
+    spanLength := &structs.BinSpanLength {
+        Num: num,
+        TimeScale: utils.TimeUnit(0),
+    }
```

Comment:
Use enum names rather than their values. So do something like
```
TimeScale: utils.TMMicrosecond
```
That makes the code more readable and would also make it so we don't get bugs with this part of the code if we ever rearrange or renumber the enum values

Reply:
Wouldn't this be dependent on the Microsecond. We were planning to use iota+1 for constants and 0 to specify if they are not set right? Or do we want this to be microsecond to denote not specified because siglens works on millisecond level?

Reply:
That's a good point. In fact I didn't notice initially that you meant this to be an invalid `TimeUnit`. Currently, 0 is mapped to `TMMicrosecond`:
```
type TimeUnit uint8

const (
	TMMicrosecond TimeUnit = iota
	TMMillisecond
	TMCentisecond
	TMDecisecond
	TMSecond
	TMMinute
	TMHour
	TMDay
	TMWeek
	TMMonth
	TMQuarter
	TMYear
)
```
You could add something like `TMInvalid` as the first one. Then in your code:
```
TimeScale: utils.TMInvaild
```
Which would make your intent more clear